### DISCUSSION TITLE
How to restricted access to multi tenancy features based on registration status

### DISCUSSION THREAD

**Discussion by devhoussam1998** [filamentphp/filament#11902] [Help]
### Package

Panel builder

### Package Version

v3.2.51

### How can we help you?

https://filamentphp.com/docs/3.x/panels/tenancy

**Comment by dissto** [comment]
Can you get into more detail. What exactly are you trying to achieve?
Do you want to restrict access to the tenant completely? Only given pages?

**Reply by devhoussam1998** [reply]
I want to restrict access to the tenancy features only for users with registration status db column cotain team not individual

**Reply by dissto** [reply]
Im still not sure what you mean by tenancy features but you can restrict access in many ways

```php
public function canAccessTenant(Model $tenant): bool
{
    return $this->teams->contains($tenant) && auth()->user()->my_db_column;
}
```

Or on a resource directly

```php
public static function canAccess(): bool
{
    return auth()->user()->my_db_column;
}
```

Etc. If you specify in more detail what you are trying to do it would be easier to help 😋

**Reply by devhoussam1998** [reply]
tenancy features are:

```
->tenant(Team::class)
->tenantRegistration(RegisterTeam::class)
->tenantProfile(EditTeamProfile::class)
```

https://filamentphp.com/docs/3.x/panels/tenancy

**Reply by dissto** [reply]
In those cases you gotta override the `canView` ?
```php
public static function canView(Model $tenant): bool
{
    return auth()->user()->my_db_column;
}
```

**Reply by devhoussam1998** [reply]
I will try it and let you know
